### PR TITLE
🌱 Bump ginkgo to v2.7.0 and sync with hack/tools

### PR DESCRIPTION
**Pull request by lentzi90**
**What this PR does / why we need it**:

Ginkgo had different versions in the root go.mod and the hack/tools/go.mod. These are now synchronized and bumped to v2.7.0.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**TODOs**:

- [x] squashed commits
- if necessary:
  - [ ] includes documentation
  - [ ] adds unit tests

/hold

### CODE REVIEW

File: api/v1alpha6/openstackmachinetemplate_webhook_test.go [hunk 1]
```diff
@@ -150,6 +150,7 @@ func TestOpenStackMachineTemplate_ValidateUpdate(t *testing.T) {
 	}
 
 	for _, tt := range tests {
+		tt := tt
```
Comment (lentzi90) [inline api/v1alpha6/openstackmachinetemplate_webhook_test.go:153]: I don't understand why this showed up now. Make lint was complaining, but for me it was complaining also on `main`, but I don't see it complaining on other PRs. :shrug:

Comment (jichenjc) [reply api/v1alpha6/openstackmachinetemplate_webhook_test.go:153]: um... this is weird .. but should not block our merge for this PR?

Comment (lentzi90) [reply api/v1alpha6/openstackmachinetemplate_webhook_test.go:153]: Yes it is fine now I was just unsure why we didn't see it before when running `make lint`. Anyway it is fixed in this PR. :slightly_smiling_face:

**Review by mdbooth — APPROVED** [review]

**Pull request merged**